File: IAIso-v5.0/core/iaiso-php/src/Core/PressureSnapshot.php
<?php

declare(strict_types=1);

namespace IAIso\Core;

/** Read-only snapshot of engine state. */
final class PressureSnapshot
{
    public function __construct(
        public readonly float $pressure,
        public readonly int $step,
        public readonly Lifecycle $lifecycle,
        public readonly float $lastStepAt,
    ) {
    }
}

File: IAIso-v5.0/core/iaiso-java/iaiso-core/src/main/java/io/iaiso/core/PressureSnapshot.java
package io.iaiso.core;

/** Read-only view of pressure-engine state at a point in time. */
public final class PressureSnapshot {
    private final double pressure;
    private final long step;
    private final Lifecycle lifecycle;
    private final double lastDelta;
    private final double lastStepAt;

    public PressureSnapshot(double pressure, long step, Lifecycle lifecycle,
                            double lastDelta, double lastStepAt) {
        this.pressure = pressure;
        this.step = step;
        this.lifecycle = lifecycle;
        this.lastDelta = lastDelta;
        this.lastStepAt = lastStepAt;
    }

    public double getPressure() { return pressure; }
    public long getStep() { return step; }
    public Lifecycle getLifecycle() { return lifecycle; }
    public double getLastDelta() { return lastDelta; }
    public double getLastStepAt() { return lastStepAt; }
}
